### ISSUE TITLE
TaskEventWatcher will not watch a folder unless IncludeSubfolders is true

### ISSUE THREAD

**Issue by dahall** [dahall/TaskScheduler#194]
Hi, I wanted to report this because the behavior in 2.5.18 is different than the library version included with the [events example](https://taskscheduler.codeplex.com/wikipage?title=EventSamples&referringTitle=Documentation) (2.5.7) and it took me a while to realize what was missing :)

That said, thanks for an amazing library!

Originally posted: 2016-04-22T07:40:49

**Comment by dahall** [comment]
What version of Windows are you running against and when constructing the TaskService, are you setting the final parameter to true?

Originally posted: 2016-04-22T08:04:33

**Comment by dahall** [comment]
Are you trying to watch a single task or all tasks in a folder? Are you setting a filter?

Originally posted: 2016-04-22T08:53:57

**Comment by dahall** [comment]
Please look at https://taskscheduler.codeplex.com/SourceControl/latest#TaskService/TestTaskService/TaskWatcherForm.cs as a full example of how this component can be used. I have tested this and cannot find the problem you've indicated.

Originally posted: 2016-04-22T09:16:44

**Comment by dahall** [comment]
I am running win7 pro sp1 x64 (spanish). I am watching a folder, no filters. This is my init code:

```
            watcher = new TaskEventWatcher();
            watcher.BeginInit();
            watcher.Folder = "\\Automation";
            watcher.IncludeSubfolders = true; // if i don't include this line, i don't get any events

            // Only watch for task events that are informational
            watcher.Filter.EventLevels = new int[]
            {(int)StandardEventLevel.LogAlways, (int) StandardEventLevel.Informational};

            // Assign an event handler for when events are recorded
            watcher.EventRecorded += Watcher_EventRecorded;

            // Start watching the folder by enabling the watcher
            watcher.Enabled = true;
            watcher.EndInit();

```
Anything else I can do to help? Thank you

Originally posted: 2016-04-22T12:54:49

**Comment by dahall** [comment]
Thank you. I found the bug. Until I release 2.5.19 (shortly) you can potentially work around the problem by adding a trailing '\' to the Folder value.

Originally posted: 2016-04-22T15:19:43

**Comment by dahall** [comment]
Great. Thank you very much.

Originally posted: 2016-04-27T14:40:07

**Comment by dahall** [comment]
Fixed in 2.5.19

Originally posted: 2016-04-27T08:31:38
